pkg/client/client.go

```diff
@@ -50,9 +51,25 @@ type tlsBootstrapClientImpl struct {
 	resource       string
 }
 
+var newLoadExecCredential = func() (*datamodel.ExecCredential, error) {
+	return loadExecCredential()
+}
+
+var newGetTlsConfig = func(pemCAs []byte, c *tlsBootstrapClientImpl, execCredential *datamodel.ExecCredential) (*tls.Config, error) {
+	return getTLSConfig(pemCAs, c.nextProto, execCredential.Spec.Cluster.InsecureSkipTLSVerify)
+}
+
+var newLoadAzureJson = func() (*datamodel.AzureConfig, error) {
+	return loadAzureJSON()
+}
+
+var newGetAuthToken = func(ctx context.Context, c *tlsBootstrapClientImpl, customClientID, resource string, azureConfig *datamodel.AzureConfig) (string, error) {
+	return c.getAuthToken(ctx, customClientID, resource, azureConfig)
```

Comment:
can't we already mock the functionality in `getAuthToken`? I'd rather not wrap method calls in functions like this if possible